sync/class.jetpack-sync-modules.php

```diff
@@ -80,16 +80,20 @@ static function initialize_modules() {
 		 */
 		$modules = apply_filters( 'jetpack_sync_modules', self::$default_sync_modules );
 
-		return array_map( array( 'Jetpack_Sync_Modules', 'initialize_module' ), $modules );
+		$modules = array_map( array( 'Jetpack_Sync_Modules', 'initialize_module' ), $modules );
+
+		return array_map( array( 'Jetpack_Sync_Modules', 'set_module_defaults' ), $modules );
 	}
 
 	static function initialize_module( $module_name ) {
-		$module = new $module_name;
+		return new $module_name;
```

Comment:
When I think of an "initialized" module, I think of a module whose initial/default values have already been set. For that reason, I think the initialize_module() method should return modules whose default values have been set. The bare object returned by this modified initialize_module() method arguably hasn't been initialized, making the method name somewhat of a misnomer (something like create_module(), that doesn't imply it's ready to use, might be more appropriate in its current state). Perhaps consider calling set_module_defaults() from within initialize_module() so that the method returns truly initialized modules, and removing the call to array_map() on line 85 that calls set_module_defaults() on all modules.